test/protocol/disconnect_dap_test.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require_relative '../support/protocol_test_case'
+
+module DEBUGGER__
+
+  class DisconnectDAPTest < ProtocolTestCase
+    PROGRAM = <<~RUBY
+       1| module Foo
+       2|   class Bar
+       3|     def self.a
+       4|       "hello"
+       5|     end
+       6|   end
+       7|   loop do
+       8|     b = 1
+       9|   end
+      10|   Bar.a
+      11|   bar = Bar.new
+      12| end
+    RUBY
+
+    def test_disconnect_without_terminateDebuggee_keeps_debuggee_alive
+      run_protocol_scenario PROGRAM, cdp: false do
+        req_dap_disconnect(terminate_debuggee: false)
+        attach_to_dap_server
+        assert_reattached
+        # suspends the debuggee so it'll take the later requests (include terminate)
+        suspend_debugee
+        req_terminate_debuggee
+      end
+    end
+
+    def test_disconnect_with_terminateDebuggee_kills_debuggee
+      run_protocol_scenario PROGRAM, cdp: false do
+        req_dap_disconnect(terminate_debuggee: true)
+      end
+    end
+
+    private
+
+    def req_dap_disconnect(terminate_debuggee:)
+      send_dap_request 'disconnect', terminateDebuggee: terminate_debuggee
+      close_reader
+    end
+
+    def suspend_debugee
+      send_dap_request "pause", threadId: 1
+    end
+
+    def assert_reattached
```

Comment:
Since we introduce `attach_to_cdp_server` and `attach_to_dap_server` as public API, we can remove them from `assert_reattach`. Therefore, here is my suggestion.

* Remove `attach_to_cdp_server` and `attach_to_dap_server` from `assert_reattach`.
https://github.com/ruby/debug/blob/master/test/support/protocol_test_case.rb#L158
* Then, we don't have to define `assert_reattached` here.

Reply:
I can do that. But I didn't define `assert_reattached` because `ProtocolTestCase` used to have it. I use it to abstract the underlying code, so the `test_disconnect_without_terminateDebuggee_keeps_debuggee_alive` test case would be easier to understand. 

I think this

```rb
attach_to_dap_server
assert_reattached
suspend_debugee
req_terminate_debuggee
```

is better than.

```rb
attach_to_dap_server
res = find_crt_dap_response
result_cmd = res.dig(:command)
assert_equal 'configurationDone', result_cmd
suspend_debugee
req_terminate_debuggee
```

Wdyt?

Reply:
I agree with that. Thank you 👍